### ISSUE TITLE
Propose document structure for markdown partials 

### ISSUE THREAD

**Issue by blackfalcon** [AlaskaAirlines/WC-Generator#323]
## Is your feature request related to a problem? Please describe.

The current document architecture for the markdown magic scenario is confusing and makes it difficult to understand where to find the right things to edit. A basic setup looks like the following. 

```
// Generated on build using README template from ./docTemplates
// Editing this file will result in it being overwritten by the template
README.md

// Root resource for code examples, this is content 
// Referenced in ./docs/partials/apiExamples.md, ./docs/partials/demo.md, ./README.md, ./docTemplates/README.md
./apiExamples
└── basic.html

// README template pulled from remote on build 
./docTemplates
└── README.md

// 
./docs
├── partials
|  ├── apiExamples.md (template)
|  ├── demo.md (template)
|  ├── description.md (content used by README and demo)
|  ├── readmeAddlInfo.md (content used for README)
|  └── useCases.md (content used by README and demo)

// the demo and apiExamples files are generated from the template located in ./docs/partials/
// Editing these files will result in them being overwritten by the templates 
./demo
├── demo.md
├── apiExamples.md
```

## Describe the solution you'd like

The following proposed re-architecture. The goal here is that there is a single place to look for code and content partials. Template documents are all centralized into a single directory versus the current 2 (./docTemplates, ./docs/partials). 

```
// Generated on build using README template from ./docs/templates/README.template
// Editing this file will result in it being overwritten by the template
README.md

// Files used for content are directly accessible from the generated file for easy reference
./demo
├── demo.md
├── apiExamples.md
├── partials
|  ├── basic.html
|  ├── description.md (content used by README and demo)
|  ├── readmeAddlInfo.md (content used for README)
|  └── useCases.md (content used by README and demo)

// All markdown template files located in this directory 
./docs
├── templates
|  ├── README.template
|  ├── apiExamples.template
|  └── demo.template
```

**Comment by jason-capsule42** [comment]
I chose to split them the way I did to reduce confusion about which files need to be updated and which are automatically generated.

When creating/managing a repo - the original file structure is such that only files under `./docs` and `./apiExamples` should be touched by the engineer. Everything outside of those two folders is a compiled document.

I would propose the following for consideration:

Stop checking this into the git repo, since we fetch the latest version every time there is no need to store it locally

> // README template pulled from remote on build 
> ./docTemplates
> └── README.md

Our docs all fall into one of two purposes:
1. local repo informational (readme.md & api.md)
1. documents for rendering on the web (demo.md & apiExamples.md)

I would like to see our folder structure better reflect that
```
_(not attached to any folder/file names used here)_

./
├── readme.md (generated)
├── api.md (generated)
├── webDocs 
|     ├── demo.md (generated)
|     └── apiExamples.md (generated)
├── apiExamples
           └── basic.html
└── docs
|     ├── readme.template
|     └── apiExamples.md (generated)
```

**Comment by geoffrich** [comment]
I'm still getting up to speed with the new docs setup, but here are some of my thoughts:

- +1 to not checking `docTemplates/README` into the repo. If this is re-fetched on every build it shouldn't be committed.
- Though along that line - if we are fetching a template from another repo, how do we handle breaking changes to that template (e.g. the kind of stuff we're talking about now, with changing folder structure)? It seems like there are filepaths hard-coded into that template.
- Is there a way that we could group all generated content (aside from the README) in a `/generated` folder? That way it would be clear what is auto-generated and what is editable.

**Comment by blackfalcon** [comment]
> +1 to not checking docTemplates/README into the repo. If this is re-fetched on every build it shouldn't be committed.

We can't do that. If it is not un version control, it will never be added to the repo and then the repo will not have a README. Even if we process this in the remote, it needs to be committed back to the repo to exist.

**Comment by blackfalcon** [comment]
> Though along that line - if we are fetching a template from another repo, how do we handle breaking changes to that template (e.g. the kind of stuff we're talking about now, with changing folder structure)? It seems like there are filepaths hard-coded into that template.

Version changes are pulled in real time with each build. The scips pull from a remote to the local and then run the build.

**Comment by geoffrich** [comment]
> We can't do that. If it is not un version control, it will never be added to the repo and then the repo will not have a README. Even if we process this in the remote, it needs to be committed back to the repo to exist.

@blackfalcon I'm specifically talking about the README _template_ in `docsTemplates`, not the generated README itself.

**Comment by blackfalcon** [comment]
In the end, the thing I am looking to improve on is the confusing mental gymnastics I need to go through to make a content change. With the current setup, I find myself having to remind myself over and over not only what version of what file to update, but where that file is. If I have to use a file search feature to find things, we are making this too hard. 

If a template and the editable file have the exact same name, e.g. demo.md, but they are in different locations, that's really confusing from the editor's perspective. 

I try hard to maintain what is going on, but I always get trapped by a case where I make an edit, and then that edit is deleted in the build because I accidentally edited the wrong file. Putting myself into the perspective of a contributor, this can get really frustrating real fast.

**Comment by blackfalcon** [comment]
> We aren't talking about not committing the generated readme.md at the root. Rather we are saying there is no need to commit the template that the readme is generated from since it's pulled from the server every time.

Ahhhh..... true facts. We should NOT be doing that for sure.

**Comment by blackfalcon** [comment]
I tried to take a stab at these changes with recent updates to Auro flight. One thing I discovered is that naming things `*.template` works just fine. Was curious about that. 

But while doing this, I also realized that the template for the README (https://raw.githubusercontent.com/AlaskaAirlines/WC-Generator/master/componentDocs/README.md) has relative paths to the documents in the directories. 

The issue here is that if we edit this template, then all the repos that currently use the current structure will break. Or we have to go back to all those reops and update the markdown docs structure and that's just not in scope. 

To address this, I propose that within the `componentDocs` dir we create `README_v2.md` and then link the updated generator to that resource.
